src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -194,16 +196,31 @@ public override void WriteObject(object graph, Type type, ODataMessageWriter mes
                 // the next page link is not set when calling WriteStart(feed) but is instead set later on that feed
                 // object before calling WriteEnd(), the next page link will be written at the end, as required for
                 // odata.streaming=true support.
-                if (nextPageLink != null)
+                if (nextLinkGenerator != null)
                 {
-                    deltaFeed.NextPageLink = nextPageLink;
+                    deltaFeed.NextPageLink = nextLinkGenerator(lastObject);
                 }
             }
 
             //End Writing of the Delta Feed
             writer.WriteEnd();
         }
 
+        /// <summary>
+        /// Creates a function that takes in an object and generates nextlink uri.
+        /// </summary>
+        /// <param name="deltaFeed">The resource set describing a collection of structured objects.</param>
+        /// <param name="enumerable">The instance representing the resourceSet being written.</param>
+        /// <param name="edmCollectionTypeReference">The EDM type of the resourceSet being written.</param>
+        /// <param name="writeContext">The serializer context.</param>
+        /// <returns>The function that generates the NextLink from an object.</returns>
+        /// <returns></returns>
+        public virtual Func<object, Uri> GetNextLinkGenerator(ODataDeltaResourceSet deltaFeed, IEnumerable enumerable, IEdmCollectionTypeReference edmCollectionTypeReference, ODataSerializerContext writeContext)
```

Comment:
>GetNextLinkGenerator [](start = 41, length = 20)

Do we have tests that validate a non-default implementation of GetNextLinkGenerator for delta feeds? since delta feeds are typically sorted by when the change occurred (which is not a property you can filter on) and each item may contain only the data that has changed, I don't think we can use our default $skiptoken implementation.

Reply:
We don't use the skiptoken implementation here for the delta feeds and we don't have tests. My change here allows for a custom delta feed serializer that only changes how the nextpage link is generated. As of now, don't change the nextlink. We can discuss this in detail in the meeting.


---
In reply to: [248346379](https://github.com/OData/WebApi/pull/1724#discussion_r248346379) [](ancestors = 248346379)